driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -475,12 +475,14 @@ private void warnOnUnsupportedOptions(final Map<String, List<String>> optionsMap
     }
 
     private void translateOptions(final Map<String, List<String>> optionsMap) {
+        boolean tlsInsecureSet = false;
```

Comment:
This implementation is really awkward but I couldn't think of a better way without a larger restructuring.